### PR TITLE
feat: add experimental Next.js, with Workers assets, template

### PR DESCRIPTION
**Pull request by petebacondarwin**
## What this PR solves / how to test

This template uses the new `@opennextjs/cloudflare` adaptor.
Fixes #000

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: experimental C3 template
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: experimental C3 template
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Changeset included
  - [ ] Changeset not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: experimental C3 template

### CODE REVIEW

File: .changeset/quiet-eels-rest.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"create-cloudflare": patch
+---
+
+feat: add experimental Next.js, with Workers assets, template
```
Comment (edmundhung) [inline .changeset/quiet-eels-rest.md:5]:
```suggestion
chore: add experimental Next.js, with Workers assets, template
```

**Review by edmundhung — APPROVED** [review]

File: .changeset/quiet-eels-rest.md (hunk 1 above)
Comment (IgorMinar) [reply .changeset/quiet-eels-rest.md:5]: this is a feature not a fix. I'm asking for an exemption from the feature freeze

File: packages/create-cloudflare/templates-experimental/next/templates/wrangler.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#:schema node_modules/wrangler/config-schema.json
+name = "<app-name>"
+main = ".worker-next/index.mjs"
+
+compatibility_date = "2024-08-29"
+compatibility_flags = ["nodejs_compat_v2"]
```
Comment (IgorMinar) [inline packages/create-cloudflare/templates-experimental/next/templates/wrangler.toml:6]: with https://github.com/cloudflare/workers-sdk/issues/6821 fixed, could we now drop the `_v2` suffix?

**Review by IgorMinar — APPROVED** [review]
LGTM, but we need a feature freeze exemption because technically this is a feature.

It's however an experimental feature with minimal blast radius, so I think this change is very safe to merge, and the reward for doing so is huge (single command to scaffold and deploy next.js apps with opennext adapter to cloudflare").

Asking for the exemption now. Please don't release this change until the exemption is granted.

PS: there is an e2e test failure, it's unclear if it's real or a flake, please take a look.

**Comment by IgorMinar** [comment]
This feature was now granted an exception from the feature freeze due to high value and low risk, see CR-1000121.

**Comment by petebacondarwin** [comment]
> PS2: could you add a minimal e2e test for this template please? thank you

I'll try but so far I have not added e2e tests for any of the experimental templates because I wasn't sure how much I would need to change the machinery.

**Review by RamIdeas — APPROVED** [review]

**Comment by IgorMinar** [comment]
Thanks for adding the test, @petebacondarwin. Excellent work!

**Pull request merged**